R/datasets.R

```diff
@@ -15,57 +15,42 @@
 #' @include schema.R
 NULL
 
-#' Reads a tabular Foundry dataset as data.frame.
+#' Reads a tabular Foundry dataset as data.frame or an arrow Table.
 #'
 #' Note that types may not match exactly the Foundry column types.
 #' See https://arrow.apache.org/docs/r/articles/arrow.html for details on type conversions
 #' from an arrow Table to a data.frame.
 #' For more advanced usage, use `read_table_arrow`.
 #'
-#' @param dataset A Dataset or a character string representing the RID or dataset path.
-#' The Foundry dataset must be tabular, i.e. have a schema.
-#' @param branch The dataset branch.
-#' @param transaction The dataset transaction.
+#' @param alias The alias representing the Dataset. It must be tabular, i.e. have a schema.
+#' @param columns The subset of columns to retrieve.
+#' @param row_limit The maximum number of rows to retrieve.
+#' @param format The output format, can be 'arrow' or 'data.frame'.
 #'
-#' @return A data.table
+#' @return A data.table or an arrow Table
 #'
 #' @export
-read_table <- function(dataset, branch = NULL, transaction = NULL) {
-  df <- read_table_arrow(dataset, branch = branch, transaction = transaction)$to_data_frame()
-  head(df, n=nrow(df))
-}
+datasets.read_table <- function(alias, columns = NULL, row_limit = NULL, format = "data.frame") { # nolint: object_name_linter
```

Comment:
So this is more like the method you'd use in transforms rather than something that exposes the full capability of the dataset API? I think we want to make sure this method matches whatever package structure exists in the Python client.